lib/HLSL/HLModule.cpp

```diff
@@ -512,6 +512,8 @@ void HLModule::LoadHLMetadata() {
           DxilMDHelper::ConstMDToUint32(options->getOperand(1)->getOperand(0)));
   }
 
+  m_pOP->SetMinPrecision(m_Options.bUseMinPrecision);
+
```

Comment:
Because minprecision isn't set.  But refreshcache would also need to be called after initializing the DxilOperation class with the correct low-precision mode.

Before, we weren't initializing HLModule from serialized intermediate state properly because we didn't set the min precision mode, but our testing wasn't sufficient to catch this.  Now, one of our tests hits a new assert, when this isn't initialized here.  I added the assert in `GetCBufferRetType` [here](https://github.com/microsoft/DirectXShaderCompiler/pull/5880/files#diff-e85a54111eb6509ee6bf740224386600db9c32fa2fc38d9b95cbb202f0882800R5082) to catch when we try to construct a type, but don't know the min precision mode, when it would make a difference.